stock_lot_create_set_reference/models/stock_picking.py

```diff
@@ -0,0 +1,19 @@
+# Copyright 2024 Quartile (https://www.quartile.co)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    def _set_auto_lot(self):
+        res = super()._set_auto_lot()
+        self.env.context = self.with_context(auto_lot_setted=True)._context
+        self.env.context = self.with_context(
+            picking_for_lot_ref={
```

Comment:
I don't see this context key being checked anywhere in the code :s

Reply:
is it necessary? then why has no test given an error? review the tests.

Reply:
I removed that context